experiment_scheduler/master/master.py

```diff
@@ -181,23 +182,26 @@ def get_available_task_managers(self):
         # return available_task_managers
         return [self.task_managers_address[0]]
 
-    def execute_task(self, task_manager):
+    async def execute_task(self, task_manager):
         """
         [TODO] add docstring
         :param task_manager:
         :return:
         """
-        prior_task = self.queued_tasks.pop(0)
-        gpu_idx = 0  # need to get idx from Resource Monitor
-        self.master_pipes[task_manager].send(
-            [
-                "run_task",
-                gpu_idx,
-                prior_task.command,
-                prior_task.name,
-                dict(prior_task.task_env),
-            ]
-        )
+        while (True):
```

Comment:
why infinite loop is needed?  _execute_command which calls execute_task also use infinite loop.